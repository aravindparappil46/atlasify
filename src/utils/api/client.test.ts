import axios from 'axios';
import {
  mockAtlassianCloudAccount,
  mockSettings,
  mockToken,
  mockUsername,
} from '../../__mocks__/state-mocks';
import type { SettingsState } from '../../types';
import {
  getAuthenticatedUser,
  getNotificationsForUser,
  markNotificationsAsRead,
  markNotificationsAsUnread,
} from './client';

jest.mock('axios');

const mockThreadId = '1234';
const mockRepoSlug = 'atlasify-app/notifications-test';

describe('utils/api/client.ts', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  describe('getAuthenticatedUser', () => {
    it('should fetch authenticated user - github', async () => {
      await getAuthenticatedUser(mockUsername, mockToken);

      expect(axios).toHaveBeenCalledWith({
        url: 'https://api.github.com/user',
        method: 'GET',
        data: {},
      });

      expect(axios.defaults.headers.common).toMatchSnapshot();
    });
  });

  describe('listNotificationsForAuthenticatedUser', () => {
    it('should list notifications for user', async () => {
      await getNotificationsForUser(
        mockAtlassianCloudAccount,
        mockSettings as SettingsState,
      );

      expect(axios).toHaveBeenCalledWith({
        url: 'https://api.github.com/notifications?participating=true',
        method: 'GET',
        data: {},
      });

      expect(axios.defaults.headers.common).toMatchSnapshot();
    });
  });

  describe('markNotificationsAsRead', () => {
    it('should mark notifications as read', async () => {
      await markNotificationsAsRead(mockAtlassianCloudAccount, [mockThreadId]);

      expect(axios).toHaveBeenCalledWith({
        url: `https://api.github.com/notifications/threads/${mockThreadId}`,
        method: 'PATCH',
        data: {},
      });

      expect(axios.defaults.headers.common).toMatchSnapshot();
    });
  });

  describe('markNotificationsAsUnread', () => {
    it('should mark repository notifications as read - github', async () => {
      await markNotificationsAsUnread(mockAtlassianCloudAccount, [
        mockThreadId,
      ]);

      expect(axios).toHaveBeenCalledWith({
        url: `https://api.github.com/repos/${mockRepoSlug}/notifications`,
        method: 'PUT',
        data: {},
      });

      expect(axios.defaults.headers.common).toMatchSnapshot();
    });
  });
});
