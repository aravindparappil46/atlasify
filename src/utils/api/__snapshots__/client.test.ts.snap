// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/api/client.ts getAuthenticatedUser should fetch authenticated user - github 1`] = `
{
  "Accept": "application/json",
  "Authorization": "Basic YXRsYXM6dG9rZW4tMTIzLTQ1Ng==",
  "Cache-Control": "no-cache",
  "Content-Type": "application/json",
}
`;

exports[`utils/api/client.ts listNotificationsForAuthenticatedUser should list notifications for user 1`] = `
{
  "Accept": "application/json",
  "Authorization": "Basic YXRsYXM6dG9rZW4tMTIzLTQ1Ng==",
  "Cache-Control": "no-cache",
  "Content-Type": "application/json",
}
`;

exports[`utils/api/client.ts markNotificationsAsRead should mark notifications as read 1`] = `
{
  "Accept": "application/json",
  "Authorization": "Basic YXRsYXM6dG9rZW4tMTIzLTQ1Ng==",
  "Cache-Control": "no-cache",
  "Content-Type": "application/json",
}
`;

exports[`utils/api/client.ts markNotificationsAsUnread should mark repository notifications as read - github 1`] = `
{
  "Accept": "application/json",
  "Authorization": "Basic YXRsYXM6dG9rZW4tMTIzLTQ1Ng==",
  "Cache-Control": "no-cache",
  "Content-Type": "application/json",
}
`;
